src/main/java/com/cloudbees/hudson/plugins/folder/computed/DefaultOrphanedItemStrategy.java

```diff
@@ -230,6 +269,27 @@ private static boolean disabled(TopLevelItem item) {
 
     @Override
     public <I extends TopLevelItem> Collection<I> orphanedItems(ComputedFolder<I> owner, Collection<I> orphaned, TaskListener listener) throws IOException, InterruptedException {
+        if (abortBuilds) {
+            Queue<Run<?, ?>> abortedBuilds = new LinkedList<>();
+            for (I item: orphaned) {
+                for (Job<?, ?> job: item.getAllJobs()) {
+                    for (Run<?, ?> build: job.getBuilds()) {
```

Comment:
Indeed you should not do this. Cut off the search when encountering a completed build. (Not totally correct, since it is possible for there to be a running build older than the last completed build, but good enough as a heuristic.) The iterator is lazy so you can keep the loop as is, so long as there is a break condition.